src/ngraph/runtime/cpu/cpu_builder.hpp

```diff
@@ -32,46 +32,101 @@
+    if (args[0].get_element_type() == element::bf16)                                               \
+    {                                                                                              \
+        auto& functors = external_function->get_functors();                                        \
+        std::function<void(void*, void*, size_t, int)> kernel;                                     \
+        kernel = OP<float>;                                                                        \
                                                                                                    \
-    auto element_count = out[0].get_size();                                                        \
-    auto arg0_buffer_index = external_function->get_buffer_index(args[0].get_name());              \
-    auto out0_buffer_index = external_function->get_buffer_index(out[0].get_name());               \
+        auto element_count = out[0].get_size();                                                    \
+        auto arg0_buffer_index = external_function->get_buffer_index(args[0].get_name());          \
+        auto out0_buffer_index = external_function->get_buffer_index(out[0].get_name());           \
                                                                                                    \
-    auto functor = [&, kernel, element_count, arg0_buffer_index, out0_buffer_index](               \
-        CPURuntimeContext* ctx, CPUExecutionContext* ectx) {                                       \
-        kernel(ctx->buffer_data[arg0_buffer_index],                                                \
-               ctx->buffer_data[out0_buffer_index],                                                \
-               element_count,                                                                      \
-               ectx->arena);                                                                       \
-    };                                                                                             \
-    functors.emplace_back(functor)
-
-#define BUILD_BINARY_ELEMWISE_FUNCTOR(OP)                                                          \
-    (void)node;                                                                                    \
-    auto& functors = external_function->get_functors();                                            \
-    std::function<void(void*, void*, void*, size_t, int)> kernel;                                  \
+        auto functor = [&, kernel, element_count, arg0_buffer_index, out0_buffer_index](           \
+            CPURuntimeContext* ctx, CPUExecutionContext* ectx) {                                   \
+            void* fp_src = std::malloc(element_count * 4);                                         \
+            void* fp_dst = std::malloc(element_count * 4);                                         \
+            ngraph::bf16_to_float(ctx->buffer_data[arg0_buffer_index], fp_src, element_count);     \
```

Comment:
This seems hacky. Instead of doing it this way, it would be better to write a graph pass to replace BF16 elementwise ops with Convert->FP32 Eltwise->Convert.

Reply:
Agree. We need #3689 for it to be a graph pass thing ...so once thats in I'll modify this.